packages/rest/src/parser.ts

```diff
@@ -134,22 +141,31 @@ function getParamFromRequest(
   request: Request,
   pathParams: PathParameterValues,
 ) {
-  let result;
   switch (spec.in) {
     case 'query':
-      result = request.query[spec.name];
-      break;
+      ensureRequestQueryWasParsed(request);
+      return request.query[spec.name];
     case 'path':
-      result = pathParams[spec.name];
-      break;
+      return pathParams[spec.name];
     case 'header':
       // @jannyhou TBD: check edge cases
-      result = request.headers[spec.name.toLowerCase()];
+      return request.headers[spec.name.toLowerCase()];
       break;
     // TODO(jannyhou) to support `cookie`,
     // see issue https://github.com/strongloop/loopback-next/issues/997
     default:
       throw RestHttpErrors.invalidParamLocation(spec.in);
   }
-  return result;
+}
+
+function ensureRequestQueryWasParsed(request: Request) {
```

Comment:
>Configure the maximum depth on per-endpoint basis depending on the parameter schema. E.g. if a parameter accepts only top-level properties R, G and B, then it's enough to set depth to 1.

How is the above done here where we're calling `qs`?

Reply:
I left that out of this initial pull request and use `qs` default configuration of `depth: 5`.

Determining the minimum depth is a bit involved, as we to visit all nested schemas and `$ref` links. And then there is the schema option `additionalProperties: true` which basically enables unlimited depth.